panic during kn service describe when a revision has ContainerMissing

<!-- If you need to report a security issue with Knative, send an email to knative-security@googlegroups.com. -->

### Bug report
```
$ kn service describe hello-example-3

Name:       hello-example-3
Namespace:  default
Age:        21h
URL:        http://hello-example-3.default.10.107.140.141.sslip.io

Revisions:
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
knative.dev/client/pkg/kn/commands/revision.WriteImage({0x2804470, 0xc000781ef0}, 0xc0002dca80)
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/revision/describe.go:165 +0x3dc
knative.dev/client/pkg/kn/commands/service.writeRevisions({0x2804470, 0xc000781cb0}, {0xc00075de70, 0x2, 0xc00099fb10}, 0x0)
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/service/describe.go:223 +0x269
knative.dev/client/pkg/kn/commands/service.describe({0x27b0480, 0xc00000e018}, 0xc0008ab400, {0xc00075de70, 0x2, 0x2}, 0x80)
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/service/describe.go:169 +0xb2
knative.dev/client/pkg/kn/commands/service.NewServiceDescribeCommand.func1(0xc000450c80, {0xc000222fb0, 0x1, 0x1})
	/home/prow/go/src/knative.dev/client/pkg/kn/commands/service/describe.go:145 +0x346
github.com/spf13/cobra.(*Command).execute(0xc000450c80, {0xc000222f80, 0x1, 0x1})
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc00045fb80)
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/prow/go/src/knative.dev/client/vendor/github.com/spf13/cobra/command.go:902
main.run({0xc00003a090, 0x3, 0x3})
	/home/prow/go/src/knative.dev/client/cmd/kn/main.go:112 +0x345
main.runWithExit({0xc00003a090, 0x22b4fe0, 0xc0000001a0})
	/home/prow/go/src/knative.dev/client/cmd/kn/main.go:52 +0x1e
main.main()
	/home/prow/go/src/knative.dev/client/cmd/kn/main.go:37 +0x50
```

<!-- Please describe what is actually happening -->

### Expected behavior
```
$ kn service describe hello-example-3

Name:       hello-example-3
Namespace:  default
Age:        21h
URL:        http://hello-example-3.default.10.107.140.141.sslip.io

Revisions:
     !  hello-example-3-00002 (latest created) [2] (7m)
        Error:  ContainerMissing
        Image:  gcr.io/knative-samples/fakeeee
  100%  @latest (hello-example-3-00001) #abcd [1] (21h)
        Image:     gcr.io/knative-samples/helloworld-go
        Replicas:  0/0

Conditions:
  OK TYPE                   AGE REASON
  !! Ready                   7m RevisionFailed
  !! ConfigurationsReady     7m RevisionFailed
  ++ RoutesReady            21h
```

<!-- Please describe what you expect to happen -->

### Steps to reproduce the problem

<!-- How can a maintainer reproduce this issue (please be detailed) -->

### kn version

<!-- Please paste the output of 'kn version' in the code block below -->

```
Version:      v1.0.0
Build Date:   2021-11-02 19:27:09
Git Revision: a1cf5bf8
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.27.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.27.0)
  - eventing.knative.dev/v1 (knative-eventing v0.27.0)
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -12,7 +12,7 @@
 | https://github.com/knative/client/pull/[#]
 ////
 
-## v1.0.1
+## v1.1.0 (Unreleased)
 [cols="1,10,3", options="header", width="100%"]
 |===
 | | Description | PR

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -12,6 +12,17 @@
 | https://github.com/knative/client/pull/[#]
 ////
 
+## v1.0.1
+[cols="1,10,3", options="header", width="100%"]
+|===
+| | Description | PR
+
+| 🐛
+| Fixed panic in kn service describe
+| https://github.com/knative/client/pull/1529[#1529]
+
+|===
+
 ## v1.0.0 (2021-11-02)
 [cols="1,10,3", options="header", width="100%"]
 |===

--- a/pkg/kn/commands/revision/describe.go
+++ b/pkg/kn/commands/revision/describe.go
@@ -162,7 +162,12 @@ func WriteImage(dw printers.PrefixWriter, revision *servingv1.Revision) {
 	// Check if the user image is likely a more user-friendly description
 	pinnedDesc := "at"
 	userImage := clientserving.UserImage(&revision.ObjectMeta)
-	imageDigest := revision.Status.ContainerStatuses[0].ImageDigest
+
+	containerStatuses := revision.Status.ContainerStatuses
+	var imageDigest string
+	if len(containerStatuses) > 0 {
+		imageDigest = containerStatuses[0].ImageDigest
+	}
 	if userImage != "" && imageDigest != "" {
 		var parts []string
 		if strings.Contains(image, "@") {

